pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go

```diff
@@ -571,3 +580,24 @@ func (cb *CollectorBundle) getTerminatedPodCollector() collectors.K8sCollector {
 	}
 	return nil
 }
+
+// hasPermission runs a single List request to check if cluster agent has permission to list the given resource
+func (cb *CollectorBundle) hasPermission(gvr schema.GroupVersionResource) bool {
+	_, err := cb.runCfg.APIClient.DynamicCl.Resource(gvr).List(context.Background(), metav1.ListOptions{})
+	return !errors.IsForbidden(err)
```

Comment:
I assume an authorized error is unlikely but thoerically possible, we could handle it here or assume the check is already running with authorized credentials but I'd rather rename the method to `isForbidden` to not draw confusion on what we are checking here